lib/core-integration/src/Test/Integration/Scenario/API/Shelley/HWWallets.hs

```diff
@@ -375,7 +400,111 @@ spec = describe "SHELLEY_HW_WALLETS" $ do
                     (#name . #getApiT . #getWalletName)
                     (`shouldBe` restoredWalletName)
                 ]
-
- where
-     restoredWalletName :: Text
-     restoredWalletName = "Wallet from pub key"
+  where
+    spec_selectCoins
+        :: (String, Maybe TxMetadata, Quantity "lovelace" Word64)
+        -> SpecWith Context
+    spec_selectCoins (testName, mTxMetadata, Quantity expectedFee) =
+        it testName $ \ctx -> runResourceT $ do
+            (w, mnemonics) <- fixtureWalletWithMnemonics (Proxy @"shelley") ctx
+            let pubKey = pubKeyFromMnemonics mnemonics
+            r <- request
+                @ApiWallet ctx (Link.deleteWallet @'Shelley w) Default Empty
+            expectResponseCode HTTP.status204 r
+            source <- restoreWalletFromPubKey
+                @ApiWallet @'Shelley ctx pubKey restoredWalletName
+            target <- emptyWallet ctx
+            let paymentCount = 1
+            targetAddresses <- take paymentCount .
+                fmap (view #id) <$> listAddresses @n ctx target
+            let targetAmounts = take paymentCount $
+                    Quantity <$> [minUTxOValue ..]
+            let targetAssets = repeat mempty
+            let payments = NE.fromList $ map ($ mempty) $
+                    zipWith AddressAmount targetAddresses targetAmounts
+            let outputs = zipWith3 ApiCoinSelectionOutput
+                    targetAddresses targetAmounts targetAssets
+            let addTxMetadata = case mTxMetadata of
+                    Nothing -> id
+                    Just txMetadata -> \(Json (Aeson.Object o)) -> Json
+                        $ Aeson.Object
+                        $ o <> ("metadata" .= (Aeson.toJSON (ApiT txMetadata)))
+            coinSelectionResponse <-
+                selectCoinsWith @n @'Shelley ctx source payments addTxMetadata
+            verify coinSelectionResponse
+                [ expectResponseCode HTTP.status200
+                , expectField #inputs
+                    (`shouldSatisfy` (not . null))
+                , expectField #outputs
+                    (`shouldSatisfy` ((Set.fromList outputs ==) . Set.fromList))
+                , expectField #change
+                    (`shouldSatisfy` (not . null))
+                ]
+            let apiCoinSelection =
+                    getFromResponse Prelude.id coinSelectionResponse
+            let fee = computeApiCoinSelectionFee apiCoinSelection
+            fee `shouldBe` Coin expectedFee
```

Comment:
Perhaps it would be good to also make a call to `postTransaction` in this test, to provide more direct confirmation that the fee is identical.

Reply:
Fixed by moving the dry-run call to `selectCoins` from `HWWallets` to `Transactions`.